R/qc_fix_helper.R

```diff
@@ -163,16 +163,38 @@ format_diff_section <- function(diff_lines) {
 }
 
 get_script_contents <- function(file_path, reference, comparator) {
-  # create copy
-  copied_file <- name_file_copy(file_path)
-  file.copy(file_path, copied_file)
-  withr::defer_parent(
-    clean_up(file_path, copied_file)
-  )
+  temp_dir <- tempdir()
+  file_diff_dir <- file.path(temp_dir, "file_diff_dir")
+  fs::dir_create(file_diff_dir)
+  withr::defer_parent({
+    if (dir.exists(file_diff_dir)) {
+      fs::dir_delete(file_diff_dir)
+    }
+  }) #, envir = globalenv()
+
+  # name the files the contents will be redirected to
+  file_at_reference <- file.path(file_diff_dir, "reference")
```

Comment:
i would instead make these random file names, so they don't clash if this script is invoked multiple times. Then you don't need the subdir at all, and can just use tempfile() invocations